onnxscript/rewriter/pattern.py

```diff
@@ -827,25 +821,22 @@ def try_rewrite(
         """If the node matches the pattern, then replace the node with the replacement pattern."""
         match = self.matches(node, model)
         if match:
-            assert match.values is not None, "Matched values should not be None."
-            if _valid_to_replace(match.values):
-                # bindings will be consumed by the replacement function
-                delta = self._replacement_pattern.get_replacement(match.bindings)
-                if len(delta.new_values) != self._target_num_outputs:
+            assert match.nodes is not None, "Matched values should not be None."
+            if _valid_to_replace(match.nodes):
+                delta = self._replacement_pattern.get_replacement(match)
+                if delta is None:
```

Comment:
```suggestion
                if replacement_subgraph is None:
```

for clarity?

Reply:
Done (not pushed yet). Minor nit: this will have to be replaced in all the places where it appears ... making it a "committable suggestion" can backfire if we push the "commit suggestion" button (since it looks like it updates only one occurrence).